packages/datadog-plugin-grpc/src/server.js

```diff
@@ -1,76 +1,74 @@
 'use strict'
 
-const Plugin = require('../../dd-trace/src/plugins/plugin')
-const { storage } = require('../../datadog-core')
-const analyticsSampler = require('../../dd-trace/src/analytics_sampler')
-const Tags = require('../../../ext/tags')
+const ServerPlugin = require('../../dd-trace/src/plugins/server')
 const { TEXT_MAP } = require('../../../ext/formats')
-const { addMethodTags, addMetadataTags, getFilter } = require('./util')
+const { addMetadataTags, getFilter, getMethodMetadata } = require('./util')
 
-class GrpcServerPlugin extends Plugin {
-  static get name () {
-    return 'http'
-  }
+class GrpcServerPlugin extends ServerPlugin {
+  static get name () { return 'grpc' }
+  static get operation () { return 'server:request' }
 
   constructor (...args) {
     super(...args)
 
-    this.addSub('apm:grpc:server:request:start', ({ name, metadata, type }) => {
-      const metadataFilter = this.config.metadataFilter
-      const store = storage.getStore()
-      const childOf = extract(this.tracer, metadata)
-      const span = this.tracer.startSpan('grpc.request', {
-        childOf,
-        tags: {
-          [Tags.SPAN_KIND]: 'server',
-          'span.type': 'web',
-          'resource.name': name,
-          'service.name': this.config.service || `${this.tracer._service}`,
-          'component': 'grpc'
-        }
-      })
-
-      addMethodTags(span, name, type)
-      addMetadataTags(span, metadata, metadataFilter, 'request')
-
-      analyticsSampler.sample(span, this.config.measured, true)
-
-      this.enter(span, store)
-    })
+    this.addTraceSub('update', ({ code }) => {
+      const span = this.activeSpan
 
-    this.addSub('apm:grpc:server:request:error', error => {
-      const store = storage.getStore()
+      if (!span) return
 
-      if (!store || !store.span) return
+      this.addCode(span, code)
+    })
+  }
 
-      this.addCode(store.span, error.code)
-      this.addError(error)
+  start ({ name, metadata, type }) {
+    const metadataFilter = this.config.metadataFilter
+    const childOf = extract(this.tracer, metadata)
+    const method = getMethodMetadata(name, type)
+    const span = this.startSpan('grpc.request', {
+      childOf,
+      service: this.config.service || `${this.tracer._service}-grpc-server`,
```

Comment:
This should just be the service name directly without a suffix (servers never had suffixes).

Reply:
You mean only do the `|| ...` bit on the client?

Reply:
Yep.